File: app/src/main/java/cz/kutner/comicsdb/connector/service/ComicsListService.java
package cz.kutner.comicsdb.connector.service;

import java.util.List;

import cz.kutner.comicsdb.model.Comics;
import retrofit.http.GET;
import retrofit.http.Query;
import rx.Observable;

public interface ComicsListService {
    @GET("/comicslist.php")
    Observable<List<Comics>> comicsList(@Query("str") int page);

    @GET("/search.php")
    Observable<List<Comics>> comicsSearch(@Query("searchfor") String keyword);

}

File: app/src/main/kotlin/cz/kutner/comicsdb/connector/service/ComicsListService.kt
package cz.kutner.comicsdb.connector.service

import cz.kutner.comicsdb.model.Comics
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface ComicsListService {
    @GET("/api.php?get=comics_list")
    fun comicsList(@Query("start") start: Int, @Query("records") records: Int): Call<List<Comics>>

    @GET("/search.php")
    fun comicsSearch(@Query("searchfor") keyword: String): Call<List<Comics>>

}
